### ISSUE TITLE
Missing documents in index due to bulk rejections

### ISSUE THREAD

**Issue by lukas-vlcek** [openshift/origin-aggregated-logging#175]
If there is a high load of records that the collector pushes to Elasticsearch then depending on collector implementation the data can be rejected (meaning Elasticsearch rejects to process specific bulk operation due to lack of internal resources). Note, this is not considered an ES error, the collector is expected to handle this correctly.

I explained details in https://github.com/ViaQ/integration-tests/issues/6
(Check out some of the later comments if you are tired of all the details in the description)

I have tested this specific issue with ViaQ code but since the configuration of Elasticsearch is the same I assume we have the problem here as well.

Basically, the solution includes:
- check that collectors can properly handle relevant HTTP response codes and act accordingly (e.g. try later)
- formulate recommendations for Elasticsearch thread pool settings that is better for logging use case

The thing is that bulk rejection is a function of several factors, at least:
- expected log message size (Java log can be larger due to ST?)
- size of the thread pool (defaults to number of cores)
- size of the bulk queue (defaults to 50)

See [threadpool docs](https://www.elastic.co/guide/en/elasticsearch/reference/1.5/modules-threadpool.html) for details.

**Comment by jcantrill** [comment]
Closing as resolved by: https://trello.com/c/GBEp8dKe/

Also ref:
https://trello.com/c/cWdPRAii/
